Dalamud/Plugin/PluginLog.cs

```diff
@@ -12,38 +14,250 @@ public static class PluginLog
         /// <summary>
         /// Log a templated message to the in-game debug log.
         /// </summary>
+        /// <param name="level">The log level to emit.</param>
         /// <param name="messageTemplate">The message template.</param>
         /// <param name="values">Values to log.</param>
-        public static void Log(string messageTemplate, params object[] values)
+        private static void Emit(LogEventLevel level, string messageTemplate, params object[] values) => Emit(level, null, messageTemplate, values);
+
+        /// <summary>
+        /// Log a templated message to the in-game debug log.
+        /// </summary>
+        /// <param name="level">The log level to emit.</param>
+        /// <param name="exception">The exception that caused the error.</param>
+        /// <param name="messageTemplate">The message template.</param>
+        /// <param name="values">Values to log.</param>
+        private static void Emit(LogEventLevel level, Exception exception, string messageTemplate, params object[] values)
         {
+#if DEBUG
+            // Due to DEBUG not inlining the below methods, we need the 2nd CallingAssembly which is the third frame.
```

Comment:
this is not guaranteed to be inline at all, even with MethodImpl

Reply:
Yeah I was looking at my own logs when dalamud was in debug and it was jacked up again. Gonna take a re-look at this at some point. I do want to expand this, at the very least for my own logging.